pymc/step_methods/hmc/base_hmc.py

```diff
@@ -52,12 +58,26 @@ class HMCStepData(NamedTuple):
     stats: dict[str, Any]
 
 
-class BaseHMC(GradientSharedStep):
+class BaseHMCState(StepMethodState):
+    adapt_step_size: bool
+    Emax: float
+    iter_count: int
+    step_size: np.ndarray
+    step_adapt: StepSizeState
+    target_accept: float
+    tune: bool
+    potential: PotentialState
+    _num_divs_sample: int
+
+
+class BaseHMC(GradientSharedStep, WithSamplingState):
```

Comment:
Which baseclass? `WithSamplingState`? Or `BaseHMCState`? If you mean the latter, it's because the `BaseHMC` step method has properties that are different from other step methods, so I need to represent its state differently. If you mean the `WithSamplingState`, that's because the `WithSamplingState` provides the `sampling_state` property accessors to the step method. Let me know if you meant something else.

Reply:
I mean all step methods should have whatever WithSamplingState implements in the base class

Reply:
Oh, I think you’re right. It must be a left over from a previous state of my commits

Reply:
I would consider a mixin to be a cleaner design, because it doesn't make third party step methods inherit `WithSamplingState` even if they aren't compatible. It should also make things easier to test by not introducing cross-dependencies.

Reply:
I don't see any compatibility issues it a step sampler spent make use of the functionality in the base class.

Reply:
Oh, I see what @michaelosthege is saying. The current design is more inline with what @ricardoV94 said, I already have a `BlockedStep` as a subclass of `WithSamplingState`, so including it in `BaseHMC` ancestors explicitly is pointless. The base `StepMethodState` only tries to access the `rng` property. I could provide a default value factory for that property, and that way there won't be any problem for third party libraries that define their own step methods. They would simply get a useless random number `Generator` object. They could use it if they wanted to though.